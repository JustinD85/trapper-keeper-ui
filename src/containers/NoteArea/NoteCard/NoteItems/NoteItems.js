import React from "react";

const NoteItems = ({ noteItems }) => (
  <div>
    {noteItems.map(item => (
      <div key={item.id} id={item.id}>
        {item.text}
      </div>
    ))}
  </div>
);

export default NoteItems;
