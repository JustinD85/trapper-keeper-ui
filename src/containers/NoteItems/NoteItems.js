import React from 'react';
import PropTypes from 'prop-types';
import { Divider, Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles';

const styles = {
  divider: {
    backgroundColor: '#ff8922'
  },
  strikethrough: {
    textDecoration: 'line-through'
  }
};

export const NoteItems = ({ noteItems, classes }) => {
  const jsxItems = noteItems.map(item => (
    <div iscompleted={JSON.stringify(item.isCompleted)} key={item.id} >
      <Typography 
        id={item.id} 
        className={item.isCompleted ? classes.strikethrough : ''} 
        style={{color: '#48494a'}}
        variant='body2'>
        {item.text}
      </Typography>
      <Divider light/>
    </div>
  ))
  return (
  <div>
    {jsxItems.filter(item => (
      item.props.iscompleted === 'false'
    ))}
    <Divider className={classes.divider}/>
    {jsxItems.filter(item => (
      item.props.iscompleted === 'true'
    ))}
  </div>
  )
};

NoteItems.propTypes = {
  noteItems: PropTypes.array
};

export default withStyles(styles)(NoteItems)