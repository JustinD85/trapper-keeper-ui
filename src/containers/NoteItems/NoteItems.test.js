import React from "react";
import { NoteItems } from "./NoteItems";
import { shallow } from "enzyme";

describe("NoteItems", () => {
  let wrapper;
  let mockItems = [
    { text: "Dylan", id: 1 ,isCompleted:true},
    { text: "Justin", id: 2 ,isCompleted:true},
    { text: "Shannon", id: 3,isCompleted:false }
  ];

  beforeEach(() => {
    wrapper = shallow(<NoteItems classes={{divider:''}} noteItems={mockItems} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
