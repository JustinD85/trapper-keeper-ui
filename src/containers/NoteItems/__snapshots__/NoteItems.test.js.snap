// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItems should match the snapshot 1`] = `
<div>
  <div
    iscompleted="false"
    key="3"
  >
    <WithStyles(Typography)
      className={false}
      id={3}
      style={
        Object {
          "color": "#48494a",
        }
      }
    >
      Shannon
    </WithStyles(Typography)>
    <WithStyles(Divider)
      light={true}
    />
  </div>
  <WithStyles(Divider)
    className=""
  />
  <div
    iscompleted="true"
    key="1"
  >
    <WithStyles(Typography)
      id={1}
      style={
        Object {
          "color": "#48494a",
        }
      }
    >
      Dylan
    </WithStyles(Typography)>
    <WithStyles(Divider)
      light={true}
    />
  </div>
  <div
    iscompleted="true"
    key="2"
  >
    <WithStyles(Typography)
      id={2}
      style={
        Object {
          "color": "#48494a",
        }
      }
    >
      Justin
    </WithStyles(Typography)>
    <WithStyles(Divider)
      light={true}
    />
  </div>
</div>
`;
